internal/summary/summary.go

```diff
@@ -24,29 +33,87 @@ func NewSummary() *Summary {
 }
 
 // Gather collates data into Summary
-func (s *Summary) Gather(services []model.Service) {
+func (s *Summary) Gather(services []model.Service) error {
 
 	for _, svc := range services {
-		var path string
 		// Assume that, in practice, there is only one mount
+		path := ""
 		if len(svc.Mounts) > 0 {
 			path = svc.Mounts[0].Repo.LocalPath
-		} else {
-			path = ""
 		}
-		s.Services = append(s.Services, &Service{
-			Name: string(svc.Name),
-			Path: path,
-		})
+
+		svcSummary := &Service{
+			Name:    string(svc.Name),
+			Path:    path,
+			k8sData: []k8sDatum{},
+		}
+
+		kubeDatum := &k8sDatum{}
+
+		entities, err := k8s.ParseYAMLFromString(svc.K8sYaml)
+		if err != nil {
+			return err
+		}
+
+		lbs := k8s.ToLoadBalancers(entities)
+
+		for _, lb := range lbs {
+			kubeDatum.LoadBalancer = append(kubeDatum.LoadBalancer, lb)
+		}
+
+		for _, e := range entities {
+			kubeDatum.Group = e.Kind.Group
+			kubeDatum.Version = e.Kind.Version
+			kubeDatum.Kinds = append(kubeDatum.Kinds, e.Kind.Kind)
+		}
+
+		svcSummary.k8sData = append(svcSummary.k8sData, *kubeDatum)
+		s.Services = append(s.Services, svcSummary)
 	}
+
+	return nil
 }
 
 func (s *Summary) Output() string {
 	ret := "\n──┤ Services Built … ├────────────────────────────────────────\n"
 
 	for _, svc := range s.Services {
-		ret += fmt.Sprintf("  • %s\n", svc.Name)
-		ret += fmt.Sprintf("    (%s)\n", svc.Path)
+		ret += fmt.Sprintf("    SERVICE NAME: %s\n", svc.Name)
+		ret += fmt.Sprintf("    WATCHING: %s\n", svc.Path)
+
+		for _, k := range svc.k8sData {
+			ret += fmt.Sprintln("    KUBERNETES INFO")
+			if len(k.Version) > 0 {
+				ret += fmt.Sprintf("      • Version: %s\n", k.Version)
+			}
+
+			if len(k.Group) > 0 {
+				ret += fmt.Sprintf("      • Group: %s\n", k.Group)
+			}
+
+			if len(k.LoadBalancer) > 0 {
+				ret += fmt.Sprintf("    LOAD BALANCER:")
+				for _, lb := range k.LoadBalancer {
+					ret += fmt.Sprintf(" %s", lb.Name)
+
+					if len(lb.Ports) > 0 {
+						ret += fmt.Sprintf(" | PORT: ")
+						for _, p := range lb.Ports {
```

Comment:
This seems like an egregious amount of code nesting. I'm open to ways of writing this that is easier to reason about!